[Tang Hau Chong] iP

hauchongtang opened the pull request:
# Duke 
> "Your mind is for having ideas, not holding them." -- David Allen [(source)](https://dansilvestre.com/productivity-quotes)

*Duke frees your mind from remembering things via a todolist feature.*

## Features
- Text based
- Easy to learn
- Fun to ~~play~~ use with

## Steps to get Duke running on your machine
1. Download the .jar file [here](https://github.com/hauchongtang/ip/releases/tag/A-Jar)
2. Double click it to run -> A terminal window will open up
3. Add your tasks
4. Have fun ! Be productive and tick off what you have done !

And it is **FREE**

### Features:
- [x] Managing Tasks
- [ ] Managing Deadlines (Coming Soon !)
- [ ] Reminders (Coming Soon !)

Commit:
Add Gradle support

hauchongtang committed:
initial commit with L-1

hauchongtang committed:
added Storage class abstraction

hauchongtang committed:
added ability to mark tasks as done/undone + Task abst

hauchongtang committed:
Added subclasses of Task -> ToDo, Deadline, Event & some exceptions handled

hauchongtang committed:
added i/o redirection technique to semi-automate testing of Duke

hauchongtang committed:
refactor exception handling in Duke class

hauchongtang committed:
added extra exception handling for string index out of range + test cases

hauchongtang committed:
Added functionality to delete task from list + test cases

hauchongtang committed:
added functionality to delete tasks + test cases

hauchongtang committed:
added StorageHandler and StorageParser classes -> ability to read tasks from .txt file

hauchongtang committed:
add early termination after checking that list is empty

hauchongtang committed:
Added StorageWriter -> Full ability to read and write to .txt file on task change

hauchongtang committed:
added datetime formatter and parser

hauchongtang committed:
Merge branch 'branch-Level-7'

hauchongtang committed:
Merge branch 'branch-Level-8'

hauchongtang committed:
resolve merge with L8

hauchongtang committed:
refactor classes into 'duke' package

hauchongtang committed:
refactor Classnames and packages

hauchongtang committed:
refactored codebase with packages & added updated tests

hauchongtang committed:
code cleanup -> remove unused vars / classes

hauchongtang committed:
Merge remote-tracking branch 'origin/add-gradle-support'

hauchongtang committed:
set up gradle and added java checkstyle support

hauchongtang committed:
initialization of junit test using gradle

hauchongtang committed:
added additional junit tests for todo, deadline, event

hauchongtang committed:
fixed all checkstyle errors except javadocs missing errors

hauchongtang committed:
added javadocs for non-trivial public methods & classes

hauchongtang committed:
Changed some class variables to private to adhere to Coding Style

hauchongtang committed:
Added find by query functionality

hauchongtang committed:
Merge branch 'branch-A-JavaDoc'

hauchongtang committed:
Merge 'branch-A-CodingStandard'

hauchongtang committed:
Added uncommited lines from merge resolution

hauchongtang committed:
Fix conflicts for merging branch Level-9 into master

hauchongtang committed:
Edit javadocs with better compliance to iP's coding standards

hauchongtang committed:
Modify naming of test methods in DukeTest class to that of junit convention

hauchongtang committed:
Added Stubs for children of Task class for the DukeTest class

hauchongtang committed:
Removed throwing of exception in constructors in Storage and Duke classes

hauchongtang committed:
Adapt javaFx to Duke with echo function + error replies

hauchongtang committed:
Merge branch 'feature/javaFxTest'

hauchongtang committed:
Refactored codebase with packages

hauchongtang committed:
Refactored Command and TaskList methods to output to javaFx interface

hauchongtang committed:
Abstracted Parser.parse method to adhere to SLAP convention + refactor code based on checkstyle guidelines

hauchongtang committed:
Configured checkstyle with gradle support

hauchongtang committed:
Merge branch 'branch-A-Checkstyle'

hauchongtang committed:
Refined javadoc comments based on guidelines & changed stubs to package private

hauchongtang committed:
Refactored Task and Command parents classes to be abstract

benjytan45678 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,46 @@
+    /**
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
```
Maybe you could add @return, @param and @throws in the java-docs for this method?

benjytan45678 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,46 @@
+
+    String getUserCommand() {
+        String nextCmdLine = scanner.nextLine();
+
```
Maybe you can remove this line of space?

benjytan45678 commented on src/main/java/duke/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * @param ui
+     * @param storage
+     * @return a result of the current deadline task execution
+     * @see duke.task.Deadline
```
You could probably add in @throws?

benjytan45678 commented on src/main/java/duke/command/InvalidCommand.java:
```diff
@@ -0,0 +1,25 @@
+     * @param taskList
+     * @param ui
+     * @param storage
+     * @return a result of the current invalid input task execution
```
You could probably add in @throws?

benjytan45678 commented on src/main/java/duke/command/MarkCommand.java:
```diff
@@ -0,0 +1,41 @@
+     * @param ui
+     * @param storage
+     * @return a result of the current MarkCommand task execution
+     * @see duke.task.TaskList
```
Missing @throws

benjytan45678 commented on src/main/java/duke/command/ToDoCommand.java:
```diff
@@ -0,0 +1,38 @@
+     * @param ui
+     * @param storage
+     * @return a result of the current to do task execution
+     * @see duke.task.ToDo
```
You could probably add in @throws?

benjytan45678 commented on src/main/java/duke/command/UnMarkCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * @param taskList
+     * @param ui
+     * @param storage
+     * @return a result of the current unmarked task execution
```
You could probably add in @throws?

benjytan45678 commented on src/main/java/duke/storage/StorageParser.java:
```diff
@@ -0,0 +1,114 @@
+
+    static Task formatTask(String rawTask) throws DukeException {
+        String identifier = rawTask.substring(0, 1).toUpperCase(Locale.ROOT);
+        boolean marked = false;
```
Maybe you could choose a better naming for boolean variable, like isMarked etc?

benjytan45678 reviewed (commented):
Overall, good job! Most of your code was easy to read and understand, except for a few minor coding standard violations such as missing @throws for the java-docs and better naming convention for some variables.

hauchongtang committed:
Add assertions to codebase

Assertions needed to ease debugging for future iterations / extension of
the codebase.

Assertions are added to ensure that objects passed into constructors are
not null.

Assertions are added to swtich cases with no clear default case.

hauchongtang committed:
Refactor codebase

Extract switch statements into shorter methods.

This will ensure clearer readibility for future iterations /
extentsions.

hauchongtang committed:
Add Streams to TaskList

Streams utilised for filtering a list of tasks to get results of
matching queries.

Using in the event that we need to do extra operations such as reduce
and sort, we can do so directly from the filter stream and then collect
the results to a list thereafter.

hauchongtang committed:
Merge pull request #1 from hauchongtang/branch-A-Assertions

Add assertions to codebase

hauchongtang committed:
Merge branch-A-CodeQuality -> master

Resolve conflicts

hauchongtang committed:
Merge branch 'branch-A-Streams'

hauchongtang committed:
Set up CICD

Automate deployment workflow

hauchongtang committed:
Remove IO redirection tests

hauchongtang committed:
Feature to view all schedules as well as view schedules within a date range

hauchongtang committed:
Add another check for the done status for Task before displaying it in specified Schedules

hauchongtang committed:
Fix import error

hauchongtang committed:
Merge branch 'branch-B-ViewSchedules'

hauchongtang committed:
Add User Guide

hauchongtang committed:
Update Readme

hauchongtang committed:
Ammend Readme

hauchongtang committed:
Add product screenshot

hauchongtang committed:
Update main README.md

hauchongtang committed:
Fix code style issues

hauchongtang committed:
Update README User Guide Docs

hauchongtang committed:
Update User Guide and README

hauchongtang committed:
Edit initial chat text

hauchongtang committed:
Fix image route on README.md
